UI and API

m0azam opened the pull request:
Please review n comment

m0azam committed:
UI and API

m0azam committed:
feat(origin-ui-core): admin user browsing

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/energywebfoundation/origin?pullRequest=916) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/energywebfoundation/origin?pullRequest=916) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: m0azam<br/>:white_check_mark: kosecki123<br/>:white_check_mark: JosephBagaric<br/>:x: Origin<br/><hr/>**Origin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/energywebfoundation/origin?pullRequest=916) it.</sub>

e00dan commented on packages/localization/src/languages/pl.json:
```diff
@@ -4,9 +4,14 @@
             "certificates": "Certyfikaty",
             "demands": "Oferty",
-            "organizations": "Organizacje"
+            "organizations": "Organizacje",
```
It's okay but no need to add corresponding keys to other languages than `en.json` for now. If translation key is not present in other language than EN, it will fallback to EN. :)

e00dan commented on packages/origin-backend-client/src/AdminClient.ts:
```diff
@@ -0,0 +1,24 @@
+import { IUser } from '@energyweb/origin-backend-core';
+import { IRequestClient, RequestClient } from './RequestClient';
+
+type onUploadProgressFunction = (progressEvent: ProgressEvent) => void;
```
Please remove this unused variable

m0azam replied:
updated

e00dan commented on packages/origin-backend-client/src/AdminClient.ts:
```diff
@@ -0,0 +1,24 @@
+type onUploadProgressFunction = (progressEvent: ProgressEvent) => void;
+
+export interface IAdminClient {
+    getAllUser(): Promise<IUser[]>;
```
This is invalid English grammar. Please use: `getAllUsers` instead. 👍

e00dan commented on packages/origin-backend-core/src/User.ts:
```diff
@@ -9,13 +9,17 @@ type Currency = string;
-    Issuer
+    Issuer,
+    Admin,
+    Agents
```
This should be singular form, so please use `Agent` instead of `Agents` as in other already existing roles. 👍

e00dan commented on packages/origin-backend/src/pods/admin/admin.controller.ts:
```diff
@@ -0,0 +1,14 @@
+
+    @Get('users')
+    @UseGuards(AuthGuard('jwt'))
+    public async getAllUser() {
```
Please use `getAllUsers` name instead of `getAllUser`.

josipbagaric committed:
Merge pull request #917 from energywebfoundation/master

Release 24/04/2020

Commit:
chore(release): publish /skip-deploy

 - @energyweb/device-registry@6.1.0
 - @energyweb/exchange-core@3.0.0
 - @energyweb/exchange-token-account@0.1.6
 - @energyweb/exchange@0.7.0
 - @energyweb/issuer@2.0.0
 - @energyweb/localization@0.3.0
 - @energyweb/migrations@3.1.0
 - @energyweb/origin-backend-app@1.1.0
 - @energyweb/origin-backend-client-mocks@2.0.1
 - @energyweb/origin-backend-client@5.0.1
 - @energyweb/origin-backend-core@2.2.0
 - @energyweb/origin-backend@5.0.1
 - @energyweb/origin-ui-core@1.1.0
 - @energyweb/origin-ui@2.1.0
 - @energyweb/solar-simulator@1.1.7
 - @energyweb/utils-general@9.0.0

josipbagaric committed:
Merge pull request #918 from energywebfoundation/release

Release to master

kosecki123 committed:
Update README.md

renovate-bot committed:
Update dependency @nestjs/schedule to v0.3.1

renovate-bot committed:
Update dependency @testing-library/react to v10.0.3

renovate-bot committed:
Update dependency @types/uuid to v7.0.3

renovate-bot committed:
Update dependency class-validator to v0.12.2

renovate-bot committed:
Update dependency concurrently to v5.2.0

renovate-bot committed:
Update dependency i18next to v19.4.3

renovate-bot committed:
Update dependency lint-staged to v10.1.7

renovate-bot committed:
Update dependency node-sass to v4.14.0

renovate-bot committed:
Update dependency pg to v8.0.3

renovate-bot committed:
Update dependency prettier to v2.0.5

renovate-bot committed:
Update dependency truffle to v5.1.23

renovate-bot committed:
Update dependency ts-node to v8.9.0

renovate-bot committed:
Update dependency yup to v0.28.4

renovate-bot committed:
Update eslint

renovate-bot committed:
Update nest monorepo to v7.0.9

renovate-bot committed:
Update webpack

renovate-bot committed:
Update dependency react-dropzone to v11

kosecki123 committed:
chore(): heroku deployments

kosecki123 committed:
Merge pull request #942 from energywebfoundation/feat/heroku-stable-deployment

chore(): heroku deployments for stable origin release

renovate-bot committed:
fix(deps): update dependency csv-parse to v4.9.0

kosecki123 committed:
Merge pull request #938 from energywebfoundation/renovate/nest-monorepo

fix(deps): update nest monorepo to v7.0.9

kosecki123 committed:
Merge pull request #923 from energywebfoundation/renovate/class-validator-0.x

fix(deps): update dependency class-validator to v0.12.2

kosecki123 committed:
Merge branch 'master' into renovate/nestjs-schedule-0.x

renovate-bot committed:
Update dependency @nestjs/swagger to v4.5.3

kosecki123 committed:
Merge pull request #931 from energywebfoundation/renovate/pg-8.x

fix(deps): update dependency pg to v8.0.3

renovate-bot committed:
Update dependency ethers to v4.0.47

kosecki123 committed:
Merge pull request #927 from energywebfoundation/renovate/concurrently-5.x

Update dependency concurrently to v5.2.0

kosecki123 committed:
Merge pull request #924 from energywebfoundation/renovate/csv-parse-4.x

Update dependency csv-parse to v4.9.0

renovate-bot committed:
Update dependency commander to v5.1.0

josipbagaric committed:
Merge pull request #936 from energywebfoundation/renovate/yup-0.x

fix(deps): update dependency yup to v0.28.4

josipbagaric committed:
Merge pull request #928 from energywebfoundation/renovate/i18next-19.x

fix(deps): update dependency i18next to v19.4.3

josipbagaric committed:
Merge pull request #920 from energywebfoundation/renovate/nestjs-swagger-4.x

Update dependency @nestjs/swagger to v4.5.3

josipbagaric committed:
Merge branch 'master' into renovate/nestjs-schedule-0.x

josipbagaric committed:
Merge pull request #925 from energywebfoundation/renovate/commander-5.x

Update dependency commander to v5.1.0

josipbagaric committed:
Merge pull request #941 from energywebfoundation/renovate/major-react

Update dependency react-dropzone to v11

josipbagaric committed:
Merge pull request #939 from energywebfoundation/renovate/webpack

Update webpack

josipbagaric committed:
Merge branch 'master' into renovate/ethers-4.x

josipbagaric committed:
chore: align ethers.js versions

josipbagaric committed:
Merge pull request #926 from energywebfoundation/renovate/ethers-4.x

Update dependency ethers to v4.0.47

josipbagaric committed:
chore: rm unused deps ws and axios

josipbagaric committed:
Merge branch 'master' into chore/rm-unused-deps

josipbagaric committed:
Merge pull request #919 from energywebfoundation/renovate/nestjs-schedule-0.x

Update dependency @nestjs/schedule to v0.3.1

m0azam committed:
Integrate with datagrid for getAllUsers API

m0azam committed:
Integrate with datagrid for getAllUsers API

josipbagaric committed:
Merge pull request #943 from energywebfoundation/chore/rm-unused-deps

chore: rm unused deps ws and axios

kosecki123 committed:
feat(origin-backend): price per device asset. initial e2e setup

kosecki123 committed:
chore(): enable origin-backend travis e2e test

kosecki123 committed:
feat(origin-backend): added automatic post for sale flag, API for updating settings

kosecki123 committed:
feat(exchange): create ask on deposit when autopostforsale is enabled

kosecki123 committed:
chore(device-registry): add autopostforsale properties

kosecki123 committed:
chore(origin-ui-core): add autopostforsale properties

kosecki123 committed:
chore(origin-backend): add missing migration run before e2e tests

kosecki123 committed:
feat(exchange): provide assetId and product with trade

kosecki123 committed:
Merge pull request #944 from energywebfoundation/feat/asset-per-device-price

Automatic ask creation using pre-defined price per device

josipbagaric committed:
fix(origin-backend): migration dropping column

josipbagaric committed:
chore(origin-backend): remove NOT NULL

josipbagaric committed:
Merge pull request #946 from energywebfoundation/fix/migration

fix(origin-backend): migration dropping column

kosecki123 committed:
chore(): heroku provision script

kosecki123 committed:
Merge pull request #947 from energywebfoundation/feat/heroku-provision-script

chore(): heroku provisioning script

kosecki123 committed:
chore(origin-ui-core); remove post for sale settings

m0azam committed:
Integrate with notification service and update API

m0azam committed:
Update sql migration

kosecki123 reviewed (changes requested):
Before the review, rebase the commits and follow the conventional commit scheme.

m0azam committed:
Merge branch 'origin/feat/PTTDEV-508-Admin-user-browsing' of https://github.com/energywebfoundation/origin into origin/feat/PTTDEV-508-Admin-user-browsing
